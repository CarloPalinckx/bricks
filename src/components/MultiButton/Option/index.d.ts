import { SFC } from 'react';
declare type PropsType = {
    isSelected: boolean;
    onClick(): void;
};
declare type StateType = {
    hovering: boolean;
};
declare const Option: SFC<PropsType>;
export default Option;
export { PropsType, StateType };
