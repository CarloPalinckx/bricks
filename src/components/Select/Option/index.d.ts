import { SFC } from 'react';
declare type PropsType = {
    label: string;
    isSelected: boolean;
    isTargeted: boolean;
    content?: JSX.Element;
    onClick(): void;
    onMouseEnter(): void;
};
declare type StateType = {
    hovering: boolean;
};
declare const Option: SFC<PropsType>;
export default Option;
export { PropsType, StateType };
