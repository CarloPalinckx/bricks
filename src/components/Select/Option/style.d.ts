import _R from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../../types/ThemeType';
declare type OptionPropsType = {
    isTargeted: boolean;
};
declare const StyledOption: _S<OptionPropsType & _R.HTMLProps<HTMLDivElement>, _T, OptionPropsType & _R.HTMLProps<HTMLDivElement>>;
export default StyledOption;
