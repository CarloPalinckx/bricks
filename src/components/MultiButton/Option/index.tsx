import React, { FunctionComponent } from 'react';
import Box from '../../Box';
import trbl from '../../../utility/trbl';
import StyledOption from './style';

type PropsType = {
    isSelected: boolean;
    onClick(): void;
};

type StateType = {
    hovering: boolean;
};

const Option: FunctionComponent<PropsType> = (props): JSX.Element => {
    /* tslint:disable:no-invalid-this */
    const onClick = props.onClick.bind(this);
    /* tslint:enable:no-invalid-this */

    return (
        <StyledOption isSelected={props.isSelected} onClick={onClick}>
            <Box padding={trbl(12, 9)}>{props.children}</Box>
        </StyledOption>
    );
};

export default Option;
export { PropsType, StateType };
