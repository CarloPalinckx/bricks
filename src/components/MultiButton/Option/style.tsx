import _R from 'react';
import styled, { withProps } from '../../../../src/utility/styled';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../../types/ThemeType';

type OptionPropsType = {
    isSelected: boolean;
};

const StyledOption = withProps<OptionPropsType, HTMLDivElement>(styled.div)`
    cursor: pointer;
    background: ${({ theme, isSelected }): string => (isSelected ? theme.MultiButton.window.secondaryColor : '')};

    &:hover {
        background: ${({ theme }): string => theme.MultiButton.window.secondaryColor};
    }
`;

export default StyledOption;
