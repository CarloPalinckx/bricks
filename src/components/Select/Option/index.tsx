import React, { SFC, MouseEvent } from 'react';
import Box from '../../Box';
import trbl from '../../../utility/trbl';
import StyledOption from './style';

type PropsType = {
    isTargeted: boolean;
    onClick(event: MouseEvent<HTMLDivElement>): void;
    onMouseEnter(): void;
};

type StateType = {
    hovering: boolean;
};

const Option: SFC<PropsType> = (props): JSX.Element => {
    /* tslint:disable:no-invalid-this */
    const onClick = props.onClick.bind(this);
    const mouseEnter = props.onMouseEnter.bind(this);
    /* tslint:enable:no-invalid-this */

    return (
        <StyledOption isTargeted={props.isTargeted} onClick={onClick} onMouseEnter={mouseEnter}>
            <Box padding={trbl(6, 18)}>{props.children}</Box>
        </StyledOption>
    );
};

export default Option;
export { PropsType, StateType };
