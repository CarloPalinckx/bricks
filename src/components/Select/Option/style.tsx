import _R from 'react';
import styled, { withProps } from '../../../../src/utility/_styled';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../../types/ThemeType';

type OptionPropsType = {
    isTargeted: boolean;
};

const StyledOption = withProps<OptionPropsType, HTMLDivElement>(styled.div)`
    cursor: pointer;
    background: ${({ theme, isTargeted }): string => (isTargeted ? theme.Select.common.secondaryColor : '')};
`;

export default StyledOption;
