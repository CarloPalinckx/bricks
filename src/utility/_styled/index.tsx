/* tslint:disable */
import React from 'react';
import * as styledComponents from 'styled-components';
import ThemeType from '../../types/ThemeType';

const {
    default: styled,
    css,
    injectGlobal,
    keyframes,
    ThemeProvider,
} = styledComponents as styledComponents.ThemedStyledComponentsModule<ThemeType>;

type StyledType = {
    theme?: ThemeType;
    className?: string;
    innerRef?: any;
};

type StyledFunction<T> = styledComponents.ThemedStyledFunction<T, ThemeType>;

function withProps<T, U extends HTMLElement = HTMLElement>(
    styledFunction: StyledFunction<React.HTMLProps<U>>,
): StyledFunction<T & React.HTMLProps<U>> {
    return styledFunction;
}

export { css, injectGlobal, keyframes, ThemeProvider, StyledType, StyledFunction, withProps };
export default styled;
