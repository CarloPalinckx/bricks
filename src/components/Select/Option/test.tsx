import React from 'react';
import { mountWithTheme } from '../../../utility/_styled/testing';
import Option from '.';
import toJson from 'enzyme-to-json';

describe('Option', () => {
    it('should render', () => {
        const component = mountWithTheme(
            <Option
                label="test"
                isSelected={true}
                isTargeted={false}
                onClick={(): void => {
                    return undefined;
                }}
                onMouseEnter={(): void => {
                    return undefined;
                }}
            />,
        );

        expect(toJson(component)).toMatchSnapshot();
    });
});
