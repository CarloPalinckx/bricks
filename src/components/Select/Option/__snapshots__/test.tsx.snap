// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should render 1`] = `
.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px 18px 6px 18px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 6px 18px 6px 18px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0px 6px 0px 0px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c0 {
  cursor: pointer;
}

.c4 {
  color: #88979d;
  font-family: Source Sans Pro,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c6 {
  color: #88979d;
  font-family: Source Sans Pro,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  height: 12px;
  width: 12px;
}

.c5 svg {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 100%;
  max-height: 100%;
  max-width: 100%;
  fill: currentColor;
}

<Option
  isSelected={true}
  isTargeted={false}
  label="test"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <styled.div
    isTargeted={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="c0"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <Box
        padding={
          Object {
            "bottom": "6px",
            "left": "18px",
            "right": "18px",
            "top": "6px",
          }
        }
      >
        <styled.div
          padding={
            Object {
              "bottom": "6px",
              "left": "18px",
              "right": "18px",
              "top": "6px",
            }
          }
        >
          <div
            className="c1"
          >
            <Box
              alignItems="center"
              inline={true}
              padding={
                Object {
                  "bottom": "6px",
                  "left": "18px",
                  "right": "18px",
                  "top": "6px",
                }
              }
            >
              <styled.span
                alignItems="center"
                inline={true}
                padding={
                  Object {
                    "bottom": "6px",
                    "left": "18px",
                    "right": "18px",
                    "top": "6px",
                  }
                }
              >
                <span
                  className="c2"
                >
                  <Box
                    inline={true}
                    margin={
                      Object {
                        "bottom": "0px",
                        "left": "0px",
                        "right": "6px",
                        "top": "0px",
                      }
                    }
                  >
                    <styled.span
                      inline={true}
                      margin={
                        Object {
                          "bottom": "0px",
                          "left": "0px",
                          "right": "6px",
                          "top": "0px",
                        }
                      }
                    >
                      <span
                        className="c3"
                      >
                        <Text
                          descriptive={true}
                        >
                          <styled.p
                            descriptive={true}
                          >
                            <p
                              className="c4"
                            >
                              <Icon
                                icon="checkmark"
                                size="small"
                              >
                                <styled.span
                                  aria-hidden={true}
                                  dangerouslySetInnerHTML={
                                    Object {
                                      "__html": "test-file-stub",
                                    }
                                  }
                                  elementSize="small"
                                  role="img"
                                >
                                  <span
                                    aria-hidden={true}
                                    className="c5"
                                    dangerouslySetInnerHTML={
                                      Object {
                                        "__html": "test-file-stub",
                                      }
                                    }
                                    role="img"
                                  />
                                </styled.span>
                              </Icon>
                            </p>
                          </styled.p>
                        </Text>
                      </span>
                    </styled.span>
                  </Box>
                  <Text
                    descriptive={true}
                    inline={true}
                  >
                    <styled.span
                      descriptive={true}
                    >
                      <span
                        className="c6"
                      >
                        test
                      </span>
                    </styled.span>
                  </Text>
                </span>
              </styled.span>
            </Box>
          </div>
        </styled.div>
      </Box>
    </div>
  </styled.div>
</Option>
`;
